File: backend/Models/ProjectDetails.cs
using System;
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;

namespace backend.Models
{
    [Table("ProjectDetails")]
    public class ProjectDetails : BaseModel
    {
        [Key]
        public int Id { get; set; }

        public string? CustomerId { get; set; }
        public string? CustomerName { get; set; }

        public string? ProjectId { get; set; }
        public string? ProjectName { get; set; }

        public int? EngagementMode { get; set; }
        public string? CurrencyType { get; set; }
        public decimal? ContractValue { get; set; }
        public decimal? EstimatedHours { get; set; }

        public DateTime? StartDate { get; set; }
        public DateTime? EndDate { get; set; }

        public string? Remarks { get; set; }
    }
}

File: frontend/src/app/master/models/projectDetails.ts
export interface ProjectDetails {
  clientId: number;
  id: number;
  customerId: string;
  customerName: string;
  projectId: string;
  projectName: string;
  engagementMode: number;
  currencyType: string;
  contractValue: number;
  startDate: Date;
  endDate: Date;
  remarks: string | null;
  active: boolean;
  status: number;
  createdOn: Date | null;
  createdBy: string | null;
  modifiedOn: Date | null;
  modifiedBy: string | null;
  estimatedHours: number;
}

export interface ProjectEngagementMode {
  id: number;
  engagementMode: string;
  remarks: string | null;
  status: number;
  createdOn: Date | null;
  createdBy: string | null;
  modifiedOn: Date | null;
  modifiedBy: string | null;
}
